rustdesk-server：增强环境变量支持

AutoCONFIG opened the pull request:
1. 实现官方文档定义的几乎完整的服务端环境变量支持
2. 移除websocket防火墙自动配置，社区版本已经不支持。此外也建议用户通过Nginx反向代理本地端口实现websocket支持
3. 默认改为非强制密钥，可配置为自定义密钥
4. luci-app-rustdesk-server需要相应的更改以适配此次更新

1715173329 commented on net/rustdesk-server/files/rustdesk-server.conf:
```diff
@@ -1,11 +1,16 @@
 	option auto_fw '0'
-	option web_client '0'
+	option key '_'
+	option port '21116'
```
rustdesk server/relay 的端口必须是连着的么？（没用过问问

AutoCONFIG replied:
改用env方式后，hbbr和hbbs读取的是同一全局“port”变量，hbbs使用这个全局port端口监听，但是hbbr会在这个全局port上自动＋1监听。这个也顺带解决了当前改用env读取端口的hbbr的监听bug

1715173329 replied:
那还是像原来那样用命令行参数传递 port？这样感觉有点 confusing

AutoCONFIG replied:
> 那还是像原来那样用命令行参数传递 port？这样感觉有点 confusing

这样也可以，但是客户端中配置ID服务器端口后，自动处理中继服务器的端口也是按照这种逻辑。尽可能的避免用户有太多的配置成本，所以不太建议
![63074ad7-9267-48e8-a333-55295ba8287a](https://github.com/user-attachments/assets/5c5dd62c-8be7-4125-948a-14d6684a5daa)

1715173329 replied:
> 这样也可以，但是客户端中配置ID服务器端口后，自动处理中继服务器的端口也是按照这种逻辑。尽可能的避免用户有太多的配置成本，所以不太建议

OK，那维持现状吧

1715173329 commented on net/rustdesk-server/files/rustdesk-server.conf:
```diff
@@ -1,11 +1,16 @@
 config relay 'hbbr'
 	option enabled '0'
-	option port '21117'
+	option downgrade_start_check '1800'
```
如果这些是上游默认值的话，建议在此处默认留空。

AutoCONFIG replied:
这些是初始默认值，留空怕引起程序运行错误（没有测试过程序是否有退回到默认值的机制，因为社区版的rustdesk-server功能做的不太完善）

1715173329 replied:
`[ -n "$hbbr_downgrade_start_check" ] && procd_append_param env DOWNGRADE_START_CHECK="$hbbr_downgrade_start_check"`

那像这样应该可以？

AutoCONFIG replied:
> `[ -n "$hbbr_downgrade_start_check" ] && procd_append_param env DOWNGRADE_START_CHECK="$hbbr_downgrade_start_check"`
> 
> 那像这样应该可以？

OK，我先试试能不能留空成功退回

1715173329 commented on net/rustdesk-server/files/rustdesk-server.init:
```diff
@@ -8,17 +8,25 @@ RUNDIR="/var/run/rustdesk-server"
+
 	config_get_bool hbbr_enabled "hbbr" "enabled" "0"
-	config_get hbbr_port "hbbr" "port" "21117"
+	config_get hbbr_downgrade_start_check "hbbr" "downgrade_start_check" "1800"
```
same here

AutoCONFIG committed:
rustdesk-server: enhanced environment variable support

- Implemented nearly complete server-side environment variable support
  as defined in the official documentation.
- Removed the automatic WebSocket firewall configuration; it is
  recommended to support WebSocket through an Nginx reverse proxy for
  local ports.
- Changed the default to non-mandatory keys, with the option to
  configure custom keys.

Signed-off-by: Yun Wang <maoerpet@foxmail.com>
